Source/NSSpeechRecognizer.m

```diff
@@ -23,8 +23,163 @@
 */
 
 #import <AppKit/NSSpeechRecognizer.h>
+#import <Foundation/NSDistantObject.h>
+#import <Foundation/NSString.h>
+#import <Foundation/NSDictionary.h>
+#import <Foundation/NSArray.h>
+#import <Foundation/NSThread.h>
+#import <Foundation/NSError.h>
+#import <Foundation/NSConnection.h>
+#import <Foundation/NSDistributedNotificationCenter.h>
 
-@implementation NSSpeechRecognizer
+#import "AppKit/NSWorkspace.h"
+
+id _speechRecognitionServer = nil;
+Class _speechRecognitionClass = nil;
+BOOL _serverLaunchTested = NO;
+
+#define SPEECH_RECOGNITION_SERVER @"GSSpeechRecognitionServer"
 
+@interface NSObject (GSSpeechRecognitionServer)
+- (NSSpeechRecognizer *)newRecognizer;
 @end
 
+@implementation NSSpeechRecognizer
+
++ (void) initialize
+{
+  _speechRecognitionClass = [NSSpeechRecognizer class];
+  _speechRecognitionServer = [[NSConnection rootProxyForConnectionWithRegisteredName: SPEECH_RECOGNITION_SERVER
+                                                                                host: nil] retain];
+  if (nil == _speechRecognitionServer)
+    {
+      NSWorkspace *ws = [NSWorkspace sharedWorkspace];
+      [ws launchApplication: SPEECH_RECOGNITION_SERVER
+                   showIcon: NO
+                 autolaunch: NO];
+    }
+}
+
+- (void) processNotification: (NSNotification *)note
+{
+  NSString *word = (NSString *)[note object];
+  NSEnumerator *en = [_commands objectEnumerator];
+  id obj = nil;
+  while((obj = [en nextObject]) != nil)
+    {
+      if ([[obj lowercaseString] isEqualToString: [word lowercaseString]])
+        {
+          [_delegate speechRecognizer: self
+                  didRecognizeCommand: word];
+        }
+    }
+}
+
+// Initialize
+- (instancetype) init
+{
+  self = [super init];
+  if (self != nil)
+    {
+      [[NSDistributedNotificationCenter defaultCenter]
+        addObserver: self
+           selector: @selector(processNotification:)
+               name: GSSpeechRecognizerDidRecognizeWordNotification
+             object: nil];
+    }
+  return self;
+}
+
++ (id) allocWithZone: (NSZone *)aZone
+{
+  if (self == _speechRecognitionClass)
+    {
+      if (nil == _speechRecognitionServer && !_serverLaunchTested)
```

Comment:
This code here belongs in the special communication method Richard talked about. This will include error handling as well as (re-)starting a missing server.